app/services/fact_importer.rb

```diff
@@ -164,7 +164,9 @@ def update_facts
     time = Time.now.utc
     updated = 0
     db_facts_names = []
-    db_facts.find_each do |record|
+    known_names = fact_names.keys
+    FactValue.joins(:fact_name).where(:host => host, 'fact_names.type' => fact_name_class_name).reorder('').find_each do |record|
```

Comment:
not crucial, but i think keeping `host.fact_values.joins...` here would be clearer.

Reply:
Okay